include/bm/bm_sim/data.h

```diff
@@ -180,28 +180,28 @@ class Data {
            typename std::enable_if<std::is_integral<T>::value, int>::type = 0>
   T get() const {
     assert(arith);
-    return static_cast<T>(value);
+    return value.convert_to<T>();
   }
 
   //! Get the value of Data has an unsigned integer
   unsigned int get_uint() const {
     assert(arith);
     // Bad ?
```

Comment:
could you remove the `Bad ?` comments (I see 3 of them) as well in this pull request?